perf-regression/gpu-margo-p2p-bw.cu

```diff
@@ -0,0 +1,796 @@
+        "\t(must be run with exactly 2 processes\n");
+
+    return;
+}
+
+/* service an RPC that runs the bandwidth test */
+static void bw_ult(hg_handle_t handle)
+{
+    int                   i;
+    bw_rpc_in_t           in;
+    bw_rpc_out_t          out;
+    ABT_thread*           tid_array;
+    struct bw_worker_arg* arg_array;
+    int                   ret;
+    double                start_time;
+    margo_instance_id     mid;
+    const struct hg_info* hgi;
+    size_t                cur_off = 0;
+    ABT_mutex             cur_off_mutex;
+    unsigned long         bytes_to_check = 0;
+    hg_size_t             x;
+
+    ABT_mutex_create(&cur_off_mutex);
+
+    /* get handle info and margo instance */
+    hgi = margo_get_info(handle);
+    assert(hgi);
+    mid = margo_hg_handle_get_instance(handle);
+    assert(mid != MARGO_INSTANCE_NULL);
+
+    ret = margo_get_input(handle, &in);
+    assert(ret == HG_SUCCESS);
+
+    tid_array = (ABT_thread *)malloc(g_opts.concurrency * sizeof(*tid_array));
+    assert(tid_array);
+    arg_array = (struct bw_worker_arg *)calloc(g_opts.concurrency, sizeof(*arg_array));
+    assert(arg_array);
+
+    start_time = ABT_get_wtime();
+    /* create requested number of workers to run transfer */
+    for (i = 0; i < g_opts.concurrency; i++) {
+        arg_array[i].start_tm           = start_time;
+        arg_array[i].mid                = mid;
+        arg_array[i].cur_off            = &cur_off;
+        arg_array[i].cur_off_mutex      = &cur_off_mutex;
+        arg_array[i].client_bulk_handle = &in.bulk_handle;
+        arg_array[i].target_addr        = &hgi->addr;
+        arg_array[i].op                 = (hg_bulk_op_t)in.op;
+        arg_array[i].duration           = in.duration;
+
+        ret = ABT_thread_create(g_transfer_pool, bw_worker, &arg_array[i],
+                                ABT_THREAD_ATTR_NULL, &tid_array[i]);
+        assert(ret == 0);
+    }
+
+    out.bytes_moved = 0;
+    for (i = 0; i < g_opts.concurrency; i++) {
+        ABT_thread_join(tid_array[i]);
+        ABT_thread_free(&tid_array[i]);
+
+        out.bytes_moved += arg_array[i].bytes_moved;
+    }
+
+    margo_respond(handle, &out);
+
+    if (in.op == HG_BULK_PULL) {
+        /* calculate how many bytes of the buffer have been transferred */
+        bytes_to_check
+            = (g_opts.g_buffer_size / g_opts.xfer_size) * g_opts.xfer_size;
+        if (out.bytes_moved < bytes_to_check) bytes_to_check = out.bytes_moved;
+
+        /* check integrity of fill pattern.  Note that this isn't as strong as
+         * checking every RDMA transfer separately since we are looping around
+         * and overwriting in a ring-buffer style.  We could corrupt early and
+         * but then overwrite it with correct results on a later pass.
+         */
+	/* Perform cudaMemcpy only if (GPU to GPU  || -k option) */
+	if (use_cuda_buf)
+    	    cudaMemcpy(g_buffer, c_buffer, g_opts.g_buffer_size, cudaMemcpyDeviceToHost);
```

Comment:
For all of the new cudaMemcpy calls in this test file, I'm guessing that if the -j and -k options aren't specified, you want to use cudaMemcpyDeviceToDevice option to make sure you're copying between GPU -> GPU between Host A and Host B.